src/action_node.cpp

```diff
@@ -182,16 +183,23 @@ NodeStatus BT::AsyncActionNode::executeTick()
             {
                 std::cerr << "\nUncaught exception from the method tick(): ["
                           << registrationName() << "/" << name() << "]\n" << std::endl;
+                // Set the exception pointer and the status atomically.
+                lock_type l(m_);
                 exptr_ = std::current_exception();
-                thread_handle_.wait();
+                setStatus(BT::NodeStatus::FAILURE);
```

Comment:
Does this have any effect? 
In this case we don't really want to return any status, just forward the exception

Reply:
My reasoning was that if the caller catchs the exception this node should still be in a "consistent" state. Without changing the state the node would be RUNNING, which to me sounds misleading. What do you think?

Reply:
I am not sure. I would FAILURE or SUCCESS have both a semantic meaning in the tree.
I would rather set it to IDLE, to communicate that the node is not RUNNING anymore, nor returned a result, but can be restarted.